test/test_python_dispatch.py

```diff
@@ -2028,6 +2028,53 @@ def __torch_dispatch__(cls, func, types, args, kwargs):
             e = SizesDefaultReturn(torch.randn(4, 2), use_wrapper_subclass)
             self.assertEqual(e.size(), (4, 2))
 
+    def test_custom_size_policy_dynamic_shapes(self):
+        data = torch.randn(6, 2)
+
+        class CustomSizeDynamicShapesTensor(torch.Tensor):
+            @staticmethod
+            def __new__(cls, inner):
+                return torch.Tensor._make_wrapper_subclass(
+                    # TODO: right now, _make_wrapper_subclass's dynamic shape interaction is not great.
+                    # Calling the overload that has kwargs causes us to go down the first overload path,
+                    # which will **always** specialize sizes.
+                    # We should probably eventually fix this so that the first overload can just handle dynamic shapes.
+                    cls,
+                    inner.size(),
+                    inner.stride(),
+                    None,
+                    None,
+                    inner.dtype,
+                    inner.layout,
+                    inner.device,
+                    False,
+                    inner.requires_grad,
+                    "sizes",
+                )
+
+            def __init__(self, inner):
+                self.inner = inner
+
+            @classmethod
+            def __torch_dispatch__(cls, func, types, args, kwargs):
+                if func == torch.ops.aten.sym_size.default:
+                    return args[0].inner.shape
+                if func == torch.ops.aten.sym_stride.default:
+                    return args[0].inner.shape
+                return NotImplemented
+
+        x = torch.ones(2, 2)
+        def trace_fn(x):
+            x_wrapper = CustomSizeDynamicShapesTensor(x)
+            return x_wrapper.size(), x_wrapper.stride()
+        fx_g = make_fx(trace_fn, tracing_mode="symbolic")(x)
+        # This is broken
```

Comment:
Is it? Seems like it does what the torch dispatch said it should do?

Reply:
**was** broken 🙂. Forgot to remove (during local testing)